src/wallet/wallet.cpp

```diff
@@ -700,8 +710,8 @@ bool CWallet::EncryptWallet(const SecureString& strWalletPassphrase)
         Lock();
         Unlock(strWalletPassphrase);
 
-        // if we are using HD, replace the HD seed with a new one
-        if (IsHDEnabled()) {
+        // if we are using HD, create a fresh seed or replace the existing seed with a new one
+        if (IsHDEnabled() && !IsWalletFlagSet(WALLET_FLAG_DISABLE_PRIVATE_KEYS)) {
```

Comment:
No. Looking at what we currently do, this should actually be checking for `HasHDSeed()`. Will change.

Reply:
So this would add a seed to an empty wallet? That should be made more clear in the documentation.

What if I want to import private keys and encrypt those? Should we add a `bool` to `encryptwallet` to determine if it should generate a key?